pkg/metrics/prometheus.go

```diff
@@ -28,6 +28,10 @@ const (
 	configLastReloadSuccessName    = metricConfigPrefix + "last_reload_success"
 	configLastReloadFailureName    = metricConfigPrefix + "last_reload_failure"
 
+	// TLS.
+	metricsTLSPrefix          = MetricNamePrefix + "tls_"
+	tlsCertsNotAfterTimestamp = metricsTLSPrefix + "cert_not_after"
```

Comment:
I think the naming should be either aligned with other providers or we should use the singular everywhere.
It looks like we used the singular in other gauges.

```suggestion
	tlsCertsNotAfterTimestamp = metricsTLSPrefix + "certs_not_after"
```